gstd/src/common/errors.rs

```diff
@@ -80,11 +82,52 @@ impl Error {
         matches!(self, Error::Timeout(..))
     }
 
-    /// Check whether an error is [`Error::ErrorReply`] and return its str
-    /// representation.
-    pub fn error_reply_str(&self) -> Option<&str> {
-        if let Self::ErrorReply(payload, _) = self {
-            payload.try_as_str()
+    /// Check whether an error is [`Error::ErrorReply`] and
+    /// return its decoded message's payload of a custom type.
+    pub fn error_reply_decoded<T: Decode>(&self) -> Option<(Result<T, Self>, ErrorReplyReason)> {
+        if let Self::ErrorReply(payload, code) = self {
+            Some((
+                T::decode(&mut payload.0.as_slice()).map_err(Error::Decode),
+                *code,
+            ))
+        } else {
+            None
+        }
+    }
+
+    /// Check whether an error is [`SimpleExecutionError::UserspacePanic`] from
+    /// error reply and return its decoded message's payload of
+    /// a custom type.
+    pub fn error_reply_panic<T: Decode>(&self) -> Option<Result<T, Self>> {
```

Comment:
looking at this Fns make me think if we should lazily load payload, only in this cases: so I'd place todo for that: Error (Self) itself doesnt need payload